From 8758f18f6627bbe66c0ec3de7914934ca71627cb Mon Sep 17 00:00:00 2001
From: Brian Ryner <bryner@google.com>
Date: Mon, 26 Jul 2021 21:35:17 +0000
Subject: [PATCH] Copy Windows API and MSVC runtime DLLs to the virtualenv.

---
 virtualenv.py | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/virtualenv.py b/virtualenv.py
index c7f71e6..23a0e58 100755
--- a/virtualenv.py
+++ b/virtualenv.py
@@ -1570,6 +1570,24 @@ def install_python(home_dir, lib_dir, inc_dir, bin_dir, site_packages, clear, sy
                 elif os.path.exists(python_dll_d_dest):
                     logger.info("Removed %s as the source does not exist", python_dll_d_dest)
                     os.unlink(python_dll_d_dest)
+
+            # Copy the api-ms-win-*.dll files as well, as they won't be found in sys.path
+            # otherwise.
+            for api_dll in glob.iglob(
+                os.path.join(os.path.dirname(sys.executable), "api-ms-win*.dll")):
+                api_dll_dest = os.path.join(
+                    os.path.dirname(py_executable), os.path.basename(api_dll))
+                logger.info("Copying API dll to %s", api_dll_dest)
+                shutil.copyfile(api_dll, api_dll_dest)
+
+            # Same for the vcruntime*.dll files.
+            for runtime_dll in glob.iglob(
+                os.path.join(os.path.dirname(sys.executable), "vcruntime*.dll")):
+                runtime_dll_dest = os.path.join(
+                    os.path.dirname(py_executable), os.path.basename(runtime_dll))
+                logger.info("Copying runtime dll to %s", runtime_dll_dest)
+                shutil.copyfile(runtime_dll, runtime_dll_dest)
+
         if IS_PYPY:
             # make a symlink python --> pypy-c
             python_executable = os.path.join(os.path.dirname(py_executable), "python")
-- 
2.33.0.309.g3052b89438-goog

