From 0dcc9b8b910d6ecd0d600e93869864d6204c7e98 Mon Sep 17 00:00:00 2001
From: Brian Ryner <bryner@google.com>
Date: Mon, 22 Mar 2021 10:32:50 +1100
Subject: [PATCH] Add define to disallow getentropy

---
 crypto/rand/rand_unix.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/crypto/rand/rand_unix.c b/crypto/rand/rand_unix.c
index ec6be791b3..c229c6ad2f 100644
--- a/crypto/rand/rand_unix.c
+++ b/crypto/rand/rand_unix.c
@@ -370,6 +370,7 @@ static ssize_t syscall_random(void *buf, size_t buflen)
      * internally. So we need to check errno for ENOSYS
      */
 #  if defined(__GNUC__) && __GNUC__>=2 && defined(__ELF__) && !defined(__hpux)
+#    if !defined(NO_GETENTROPY)
     extern int getentropy(void *buffer, size_t length) __attribute__((weak));
 
     if (getentropy != NULL) {
@@ -378,6 +379,7 @@ static ssize_t syscall_random(void *buf, size_t buflen)
         if (errno != ENOSYS)
             return -1;
     }
+#    endif
 #  else
     union {
         void *p;
-- 
2.31.0.rc2.261.g7f71774620-goog

