--- pyobjc-framework-libdispatch-7.3.orig/pyobjc_setup.py	2021-06-05 20:25:38.000000000 +1000
+++ pyobjc-framework-libdispatch-7.3/pyobjc_setup.py	2021-07-21 02:26:19.971571688 +1000
@@ -446,17 +446,18 @@
             cflags.append(data)
             cflags.append(
                 "-DPyObjC_BUILD_RELEASE=%02d%02d"
-                % (tuple(map(int, version.split("."))))
+                % (tuple(map(int, version.split(".")[:2])))
             )
         else:
             cflags.append(
                 "-DPyObjC_BUILD_RELEASE=%02d%02d"
-                % (tuple(map(int, os_level.split("."))))
+                % (tuple(map(int, os_level.split(".")[:2])))
             )
 
     else:
         cflags.append(
-            "-DPyObjC_BUILD_RELEASE=%02d%02d" % (tuple(map(int, os_level.split("."))))
+            "-DPyObjC_BUILD_RELEASE=%02d%02d"
+            % (tuple(map(int, os_level.split(".")[:2])))
         )
 
     if os_level == "10.4":
